### PR TITLE
docs: an example to extract Cloud Batch logs

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11935)

### CODE REVIEW

File: examples/batch_logging.cc [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    throw std::runtime_error("Error parsing Job description");
+  }
+  // Create a client and issue the request.
+  return client.CreateJob(request).value();
+}
+
+void WaitForJob(google::cloud::batch_v1::BatchServiceClient client,
+                std::string job_name) {
+  std::cout << "\nWaiting for " << job_name << std::flush;
+  // It taskes about 60 seconds to finish a job, so waiting for about 5 minutes
```
Comment (devbww) [inline examples/batch_logging.cc:119]: s/taskes/takes/

File: examples/batch_logging.cc [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  std::cout << "Created test job: " << job.name() << "\n";
+
+  // Wait until the job completes, otherwise the logs may be empty. The logs may
+  // still be delayed, but this is less likely.
+  WaitForJob(client, job.name());
+
+  std::cout << "Running JobLogs() test" << std::endl;
+  JobLogs({project_id, location_id, job_id});
+
+  //  (void) client.DeleteJob(job.name()).get().value();
```
Comment (devbww) [inline examples/batch_logging.cc:169]: Commented out?

File: examples/batch_logging.cc (hunk 1 above)
Comment (coryan) [reply examples/batch_logging.cc]: Fixed here and in the cut&paste source.

File: examples/batch_logging.cc (hunk 2 above)
Comment (coryan) [reply examples/batch_logging.cc]: Fixed.  I needed to debug something and forgot to restore this.

**Review by devbww — APPROVED** [review]

**Pull request merged**